src/libsyntax/util/parser.rs

```diff
@@ -106,7 +106,8 @@ impl AssocOp {
             Token::OrOr => Some(LOr),
             Token::DotDot => Some(DotDot),
             Token::DotDotEq => Some(DotDotEq),
-            Token::DotDotDot => Some(DotDotEq), // remove this after SNAP
+            // DotDotDot is no longer supported, but we need some way to display the error
+            Token::DotDotDot => Some(DotDotEq),
```

Comment:
My main concern is that in order to show the error we need to parse the expression, so here I'm parsing `Token::DotDotDot` as `AssocOp::DotDotEq`, and in src/libsyntax/parse/parser.rs I check for `Token::DotDotDot` and `AssocOp::DotDotEq` to show the error. An alternative would be returning `None` here but then the expression would be incomplete (adding the generic "expected one of" error).

Reply:
>I'm parsing `Token::DotDotDot` as `AssocOp::DotDotEq`

That's exactly what is needed for good error recovery.

Reply:
Nice, thanks!